system/Database/OCI8/Builder.php

```diff
@@ -255,6 +255,8 @@ protected function _updateBatch(string $table, array $keys, array $values): stri
                 return ''; // @codeCoverageIgnore
             }
 
+            // remove any constraints from updateFields
+            $this->updateFields($this->QBOptions['updateFields'] ?? [], false, $constraints);
```

Comment:
> I found a bug where a field that exists in constraints and also is in updateFields causes and error. This is only for Oracle.

This bug seems to be also in `develop`.
Can you send another PR to `develop`?

Reply:
$builder->onConstraint('email')->updateFields(['email', 'name'])->updateBatch($data);

If only happens when you explicitly define with updateFields(). If you let the system define them based on the data then everything is fine. So it is kind of a bug but not really because you shouldn't be defining a constraint as an updateField.